ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java

```diff
@@ -320,28 +316,32 @@ private void loadTables() {
     // load tables from catalog
     Set<TableIdentifier> tableIdentifiers = com.netease.arctic.ams.server.utils.CatalogUtil.loadTablesFromCatalog();
 
-    Tasks.foreach(tableIdentifiers)
-        .suppressFailureWhenFinished()
-        .noRetry()
-        .executeWith(getProcessTablesThreadPool())
-        .run(tableIdentifier -> {
-          List<OptimizeTaskItem> tableOptimizeTasks = optimizeTasks.remove(tableIdentifier);
-
-          TableMetadata tableMetadata = buildTableMetadata(tableIdentifier);
-          TableOptimizeItem arcticTableItem = new TableOptimizeItem(null, tableMetadata);
-          TableOptimizeRuntime oldTableOptimizeRuntime = tableOptimizeRuntimes.remove(tableIdentifier);
-          arcticTableItem.initTableOptimizeRuntime(oldTableOptimizeRuntime)
-              .initOptimizeTasks(tableOptimizeTasks);
-          if (CompatiblePropertyUtil.propertyAsBoolean(tableMetadata.getProperties(),
-              TableProperties.ENABLE_SELF_OPTIMIZING,
-              TableProperties.ENABLE_SELF_OPTIMIZING_DEFAULT)) {
-            addTableIntoCache(arcticTableItem, tableMetadata.getProperties(), oldTableOptimizeRuntime == null);
-          } else {
-            unOptimizeTables.add(tableIdentifier);
-            persistTableRuntime(arcticTableItem.getTableOptimizeRuntime());
-          }
-        });
-    tablesSvc.shutdown();
+    ExecutorService executorService = Executors.newFixedThreadPool(loadTablePoolSize());
+    try {
+      Tasks.foreach(tableIdentifiers)
```

Comment:
load tables task can also process by `parallelProcessTables` then move inited check to head of functions: `addNewTables`, `clearRemovedTables`, `addOptimizingTables`, `clearUnoptimizingTables`